setup.py

```diff
@@ -594,6 +605,57 @@ def build_package() -> int:
     size = du(src_xz)
     print(f"{src_xz}: {size}B")
 
+    if is_Arch():
+        if os.geteuid() == 0:
+            raise RuntimeError("Arch packaging must be run as a regular user (makepkg refuses to run as root)")
+        makepkg = shutil.which("makepkg")
+        if not makepkg:
+            raise RuntimeError("cannot find `makepkg` (install the `pacman`/`pacman-makepkg` tooling)")
+
+        arch_dir = os.path.join("build", f"archpkg-{XPRA_VERSION}")
+        os.makedirs(arch_dir, mode=0o755, exist_ok=True)
+        tar_name = f"xpra-{XPRA_VERSION}.tar.xz"
+        shutil.copy(src_xz, os.path.join(arch_dir, tar_name))
+        pkg_arch = os.uname().machine
+        with open(os.path.join(arch_dir, "PKGBUILD"), "w", encoding="utf-8") as f:
+            f.write("""# Generated by `python3 ./setup.py package`
+pkgname=xpra
+pkgver={pkgver}
+pkgrel=1
+pkgdesc='Multi-platform remote display server and client'
+arch=('{pkg_arch}')
+url='https://xpra.org/'
+license=('GPL2')
+
+depends=('python')
+makedepends=('gcc' 'make' 'binutils' 'patch' 'fakeroot' 'pkgconf' 'cython' 'python-setuptools')
+
+source=('{tar_name}')
+sha256sums=('SKIP')
+
+build() {{
+  cd "$srcdir/xpra-$pkgver"
+  python3 ./setup.py build
+}}
+
+package() {{
+  cd "$srcdir/xpra-$pkgver"
+  python3 ./setup.py install --root="$pkgdir" --prefix=/usr --skip-build --optimize=1
+}}
+""".format(pkgver=XPRA_VERSION, tar_name=tar_name, pkg_arch=pkg_arch))
```

Comment:
One could make the case for storing this template in https://github.com/Xpra-org/xpra/tree/master/packaging, which would make it easier to find.
Especially since this build file is humongous.